app/controllers/challenge_signups_controller.rb

```diff
@@ -322,4 +323,54 @@ def prompt_meme_to_csv
 
     csv_array
   end
+
+  private
+
+  def challenge_signup_params
+    params.require(:challenge_signup).permit(
+      :pseud_id,
+      requests_attributes: nested_prompt_params,
+      offers_attributes: nested_prompt_params
+    )
+  end
+
+  def nested_prompt_params
+    [
+      :id,
+      :collection_id,
+      :title,
+      :url,
+      :any_fandom,
+      :any_character,
+      :any_relationship,
+      :any_freeform,
+      :any_category,
+      :any_rating,
+      :any_warning,
+      :anonymous,
+      :description,
+      :_destroy,
+      tag_set_attributes: [
+        :id,
```

Comment:
So I've looked into this, and here are my thoughts:

The reason I included `:id` in the PromptsController (and the ChallengeSignupsController) is because it gets sent through by the params, and I whitelisted all of the params that I saw come through by using `binding.pry` to inspect them in various situations. Also, Rails passes through the primary key for nested attributes by default.

However, I've edited a prompt with existing tag sets, and while the id is present under params[:prompt][:tag_set_attributes], it is quietly excluded in the prompt_params method instead of raising a ForbiddenAttributesError. I thought that might be because TagSet doesn't have attributes protection on it, so I added that to see what would happen - and the behavior is the same. So for whatever reason, the protected attributes gem doesn't seem to care about the nested attributes' `:id`. 

So I would say that even though everything's passing without the id in the PromptsController, the fact that it does come in as a param maybe suggests that there's no reason *not* to allow it. I'm pretty stumped as to why it doesn't raise, though. :woman_shrugging: